File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02550/p02550_num2_parsed.py
N, X, M = map(int, input().split())

# calculate the remainder of x by m
def f(x, m):
    return x % m

A = [X]  # initialize the sequence A
for i in range(1, N):
    A.append(f(A[i-1]**2, M))  # compute A_i using the recurrence relation

sum_A = sum(A)  # calculate the sum of the sequence A
print(sum_A)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02550/p02550_num2_parsed.cpp
#include <iostream>
using namespace std;

int main() {
    // Read the input values for N, X, and M
    int N, X, M;
    cin >> N >> X >> M;

    // Calculate the sequence A
    int A = X;
    int sum = A;
    for (int i = 2; i <= N; i++) {
        A = (A * A) % M; // Apply the recurrence relation
        sum += A;
    }

    // Print the sum
    cout << sum << endl;

    return 0;
}